lib/reboot_cmds.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+
+"""
+Some uaclient operations cannot be fully completed by running a single
+command. For example, when upgrading uaclient from trusty to xenial,
+we may have a livepatch change in the contract, allowing livepatch to be
+enabled on xenial. However, during the upgrade we cannot install livepatch on
+the system, only after a reboot.
+
+To allow uaclient to postpone commands that need to be executed in a system boot,
+we are using this script, which will basically try to reprocess contract deltas
+on boot time.
+"""
+import logging
+import os
+import sys
+
+from uaclient import config
+
+from uaclient.util import subp, ProcessExecutionError
+from uaclient.cli import setup_logging, assert_lock_file
+
+
+BASE_PATH = "/var/lib/ubuntu-advantage/"
+REBOOT_CMD_MARKER_FILE = BASE_PATH + "marker-reboot-cmds-required"
+
+
+def run_command(cmd):
+    try:
+        out, _ = subp(cmd.split())
+        logging.debug("Successfully executed cmd: {}".format(cmd))
+    except ProcessExecutionError as exec_error:
+        msg = (
+            "Failed running cmd: {}\n"
+            "Return code: {}\n"
+            "Stderr: {}\n"
+            "Stdout: {}".format(
+                cmd, exec_error.exit_code, exec_error.stderr, exec_error.stdout
+            )
+        )
+        logging.debug(msg)
```

Comment:
do we want to remove REBOOT_CMD_MARKER_FILE here? or Leave it in place to retry on next reboot?

Reply:
That's a good question. Ideally, this decision should be based on the error context. For example, on network error issues, I do believe that we should maintain the marker file. However, I don't know how easy it would be to have that granularity in the code.

I believe that, initially, we could treat this operation as an one-shot attempt to get the system in expected state. If we fail, we can abort.

But if you belive we can treat well known errors, I am okay with that as well